File: src/test/java/j/examples/rules/feedback/PdfExtensionFileRuleTest.java
package j.examples.rules.feedback;

import org.junit.jupiter.api.Test;

import java.io.File;

import static j.examples.rules.feedback.AssertionUtils.assertNotValid;
import static j.examples.rules.feedback.AssertionUtils.assertValid;

public class PdfExtensionFileRuleTest {
    private static final PdfExtensionFileRule rule = new PdfExtensionFileRule();

    @Test
    void files_without_extension_are_rejected() {
        assertNotValid(() -> rule.validate(new File("invoice")),
            "The file invoice is not a .pdf file");
    }

    @Test
    void files_with_extension_pdf_are_accepted() {
        assertValid(() -> rule.validate(new File("invoice.pdf")));
    }

    @Test
    void files_with_extension_other_than_pdf_are_rejected() {
        assertNotValid(
            () -> rule.validate(new File("invoice.doc")),
            "The file invoice.doc is not a .pdf file"
        );
    }
}

File: src/test/kotlin/k/examples/rules/guarding/PdfExtensionFileRuleTest.kt
package k.examples.rules.guarding

import k.examples.rules.guarding.AssertionUtils.assertNotValid
import k.examples.rules.guarding.AssertionUtils.assertValid
import org.junit.jupiter.api.Test
import java.io.File


class PdfExtensionFileRuleTest {
    private val rule = PdfExtensionFileRule()

    @Test
    fun `Files with extension pdf are accepted`() {
        assertValid { rule.validate(File("invoice.pdf")) }
    }

    @Test
    fun `Files with extension other than pdf are rejected`() {
        assertNotValid(
            { rule.validate(File("invoice.doc")) },
            """The file invoice.doc is not a .pdf file"""
        )
    }
}